server/dashboard.go

```diff
@@ -70,20 +71,45 @@ func (svr *Service) RunDashboardServer(address string) (err error) {
 		http.Redirect(w, r, "/static/", http.StatusMovedPermanently)
 	})
 
-	server := &http.Server{
-		Addr:         address,
-		Handler:      router,
-		ReadTimeout:  httpServerReadTimeout,
-		WriteTimeout: httpServerWriteTimeout,
+	if svr.cfg.DashboardTLSMode {
```

Comment:
We needn't fullfill all variables.

```go
    server := &http.Server{
        Addr:         address,
        Handler:      router,
        ReadTimeout:  httpServerReadTimeout,
        WriteTimeout: httpServerWriteTimeout,
    }
    ln, err := net.Listen("tcp", address)
    if err != nil {
        return err
    }

    if svr.cfg.DashboardTLSMode {
        cert, err := tls.LoadX509KeyPair(svr.cfg.DashboardTLSCertFile, svr.cfg.DashboardTLSKeyFile)
        if err != nil {
            return err
        }
        tlsCfg := &tls.Config{
            Certificates: []tls.Certificate{cert},
        }
        ln = tls.NewListener(ln, tlsCfg)
    }
    go server.Serve(ln)
    return
```

Reply:
Resolved. Also Tried to create `tls.NewListener` with the HTTP listener but since it tries to direct HTTPS connections to HTTP host which gives the following error: `Client sent an HTTP request to an HTTPS server.`

Reply:
Now go-version-last gives an error but I couldn't figured out why...

Reply:
I'm not sure why you create a new http server for that. Just reuse the same one.

SSL termination has been down in tls listener.

Reply:
ReadTimeout and WriteTimeout gives error on CircleCI, refer to my previous reply.

Reply:
I don't know what's your meaning.

You'd better solve the problem, not ignore the configuration.

Reply:
Okay, in CircleCI run #1103, can you guide me what went wrong on it?

Reply:
I think it's a temporary error. Retry the test can fix it.

Reply:
Simplified both HTTP and HTTPS into single server and CircleCI didn't throw error this time. You can check it.